frontend/src/components/StockNews/StockNews.jsx

```diff
@@ -1,10 +1,62 @@
 import "./StockNews.css";
+import { useState, useEffect, useContext } from "react";
+import axios from "axios";
+import { formatDate, firstWord } from "../../utils.js";
+import { UserContext } from "../App/App";
 
-export default function StockNews() {
+export default function StockNews({ stocks }) {
+    const [stockNews, setStockNews] = useState([]);
+    const [currentStock, setCurrentStock] = useState(null);
+    const { setError } = useContext(UserContext);
+
+    useEffect(() => {
+        const fetchStockNews = async () => {
+            if (currentStock) {
+                try {
+                    const searchTerm = firstWord(currentStock.name);
+                    const url = `https://newsapi.org/v2/everything?q=${searchTerm}&sortBy=relevancy&pageSize=5&apiKey=${import.meta.env.VITE_NEWS}`;
+                    const response = await axios.get(url);
+                    setStockNews(response.data.articles);
+                } catch (error) {
+                    setError(`${error.message}: Please try again later.`);
+                }
+            }
+        };
+        fetchStockNews();
+    }, [currentStock]);
 
     return (
         <div className="stock-news">
             <h1>Related News</h1>
+            <select className="stock-options" value={currentStock?.ticker || ""} onChange={(event) => setCurrentStock(JSON.parse(event.target.value))}>
+                <option value="" disabled defaultValue>Select a Stock You Follow</option>
+                {
+                    stocks?.map((stock) => (
+                        <option key={stock.ticker} value={JSON.stringify(stock)}>{stock.ticker}</option>
+                    ))
+                }
+            </select>
+            <div className="news">
+                {stockNews?.map((article) => (
+                    <div className="article-margin" key={article.url}>
+                        <a href={article.url} target="_blank" rel="noopener noreferrer" className="article-link">
+                            <div className="article">
+                                <div className="article-logo">
+                                    <img
+                                        src={currentStock.logo}
+                                        alt=""
```

Comment:
nit: good idea to add `alt` to be text description (could use "< current stock > logo" for accessibility (also sometimes referred to as a11y). It's helpful for people who may use screenreaders/Voiceovers. [Web accessibility](https://en.wikipedia.org/wiki/Web_Content_Accessibility_Guidelines) is a pretty neat field to learn about best practices in general and how someone with disabilities may interact with a page :)

Reply:
ah noticed you added it in the other commit:)